test/phoenix/digester_test.exs

```diff
@@ -172,38 +179,46 @@ defmodule Phoenix.DigesterTest do
       input_path = Path.join("tmp", "phoenix_digest_twice")
       input_file = Path.join(input_path, "file.js")
 
-      File.rm_rf!(input_path)
-      File.mkdir_p!(input_path)
-      File.mkdir_p!(@output_path)
+      try do
+        File.rm_rf!(input_path)
+        File.mkdir_p!(input_path)
+        File.mkdir_p!(@output_path)
 
-      File.write!(input_file, "console.log('test');")
-      assert :ok = Phoenix.Digester.compile(input_path, @output_path, true)
+        File.write!(input_file, "console.log('test');")
+        assert :ok = Phoenix.Digester.compile(input_path, @output_path, true)
 
-      json1 = Path.join(@output_path, "cache_manifest.json") |> json_read!()
-      assert Enum.count(json1["digests"]) == 1
+        json1 = Path.join(@output_path, "cache_manifest.json") |> json_read!()
+        assert Enum.count(json1["digests"]) == 1
 
-      File.write!(input_file, "console.log('test2');")
-      assert :ok = Phoenix.Digester.compile(input_path, @output_path, true)
+        File.write!(input_file, "console.log('test2');")
+        assert :ok = Phoenix.Digester.compile(input_path, @output_path, true)
 
-      json2 = Path.join(@output_path, "cache_manifest.json") |> json_read!()
-      assert Enum.count(json2["digests"]) == 2
+        json2 = Path.join(@output_path, "cache_manifest.json") |> json_read!()
+        assert Enum.count(json2["digests"]) == 2
+      after
+        File.rm_rf!(input_path)
+      end
     end
 
     test "doesn't duplicate files when digesting and compressing twice" do
       input_path = Path.join("tmp", "phoenix_digest_twice")
       input_file = Path.join(input_path, "file.js")
 
-      File.rm_rf!(input_path)
-      File.mkdir_p!(input_path)
-      File.write!(input_file, "console.log('test');")
-
-      assert :ok = Phoenix.Digester.compile(input_path, input_path, true)
-      assert :ok = Phoenix.Digester.compile(input_path, input_path, true)
-
-      output_files = assets_files(input_path)
-      refute "file.js.gz.gz" in output_files
-      refute "cache_manifest.json.gz" in output_files
-      refute Enum.any?(output_files, &(&1 =~ ~r/file-#{@hash_regex}.[\w|\d]*.[-#{@hash_regex}/))
+      try do
+        File.rm_rf!(input_path)
+        File.mkdir_p!(input_path)
+        File.write!(input_file, "console.log('test');")
+
+        assert :ok = Phoenix.Digester.compile(input_path, input_path, true)
+        assert :ok = Phoenix.Digester.compile(input_path, input_path, true)
+
+        output_files = assets_files(input_path)
+        refute "file.js.gz.gz" in output_files
+        refute "cache_manifest.json.gz" in output_files
+        refute Enum.any?(output_files, &(&1 =~ ~r/file-#{@hash_regex}.[\w|\d]*.[-#{@hash_regex}/))
+      after
+        File.rm_rf!(input_path)
```

Comment:
Do we need to clean up before and after? Only before should be enough, no? It doesn’t hurt to do both, so this is fine, but maybe doing less would work too!